config.md

```diff
@@ -146,8 +146,8 @@ Runtimes MUST/SHOULD/MAY implement the following option strings for Linux:
  `sync`           | MUST        | [^1]
  `tmpcopyup`      | MAY         | copy up the contents to a tmpfs
  `unbindable`     | MUST        | [^2] (bind mounts)
- `idmap`          | SHOULD      | Indicates that the mount has `uidMappings` and `gidMappings` specified. This option SHOULD NOT be passed to the underlying [`mount(2)`][mount.2] call. If supported, the runtime MUST return an error if this option is provided and either of `uidMappings` or `gidMappings` are empty or not present.
```

Comment:
The first sentence for both flags is not correct for the crun case -- I think we will need to do something more like:

1. `idmap` and `ridmap` indicate that the mount must have an idmapping applied.
   - If `uidMappings` or `gidMappings` are specified, the runtime MUST use those.
   - If `uidMappings` or `gidMappings` are not specified, the runtime MAY (SHOULD? MUST?) use the container's user namespace mapping.
   - If `idmap` or `ridmap` are specified but there are no mapping specified on the mount and the container isn't using user namespaces, an error MUST (SHOULD?) be returned.
2. The current text for `uidMappings` and `gidMappings` doesn't specify what technology should be used -- this should probably also be changed so that (1) can more coherently say what kind of mapping is needed.

Reply:
thanks, I've amended the patch with your suggestions.

About point 2, I am not sure, could the runtime decide to achieve it using a different technology (e.g. seccomp notify)?

Reply:
Yeah, I suspect that not specifying it is _fine_ (you could even do it with FUSE or something similarly fruity) but at the moment the text of `uidMappings` and `gidMappings` has basically no explanation of what "mapping" actually means in this context. Maybe we can just have a reference to `MOUNT_ATTR_IDMAP` and say that the behaviour needs to match but the exact technology used is not mandated? Or say that runtimes SHOULD use `MOUNT_ATTR_IDMAP`?

(Then again, I suspect most users would expect only `MOUNT_ATTR_IDMAP` to be used and not using it would be something that could trip up container guests.)

Reply:
good idea, we are already requiring that for `uidMappings and `gidMappings`.

Amended and pushed a new version